src/renderer/components/NodeDocumentation/NodeDocs.tsx

```diff
@@ -30,61 +30,98 @@ import { NodeExample } from './NodeExample';
 import { SchemaLink } from './SchemaLink';
 
 interface InputOutputItemProps {
-    label: string;
-    description?: string | null;
+    item: Input | Output;
     type: Type;
-    optional: boolean;
-    hasHandle?: boolean;
 }
-const InputOutputItem = memo(
-    ({ label, description, type, optional, hasHandle = true }: InputOutputItemProps) => {
-        if (optional) {
-            // eslint-disable-next-line no-param-reassign
-            type = withoutNull(type);
-        }
 
-        const handleColors = getTypeAccentColors(type);
+const InputOutputItem = memo(({ type, item }: InputOutputItemProps) => {
+    if ('optional' in item && item.optional) {
+        // eslint-disable-next-line no-param-reassign
+        type = withoutNull(type);
+    }
 
-        return (
-            <ListItem my={2}>
-                <HStack>
+    const handleColors = getTypeAccentColors(type);
+
+    const isFileInput = 'kind' in item && item.kind === 'file';
+    const supportedFileTypes = isFileInput ? item.filetypes : [];
+    const isPrimaryInput = isFileInput && 'primaryInput' in item && item.primaryInput;
```

Comment:
Why do you even need to do these `in` checks?

Reply:
TS complains. See this: https://stackoverflow.com/questions/58974640/typescript-property-does-not-exist-on-union-type

Basically, since its a union, .optional, .kind, etc don't exist on outputs and therefore TS yells at you if you don't type narrow first

Reply:
Hence my question. Both `Input` and `Output` have a `kind` property, so this check shouldn't be necessary. And the `'primaryInput' in item` also shouldn't be necessary because TS already narrowed it down to FileInput because of the `isFileInput &&`.

Reply:
Hmm lemme try

Reply:
yeah it's fine without them. I guess this is one of those cases where whatever way I was originally doing it required something like that, and when I split it up and moved things I kept some unnecessary things